File: Medium/2025-10-16-2598-Smallest-Missing-Non-negative-Integer-After-Operations/solution.java
import java.util.HashSet;

class Solution {
    public int findSmallestInteger(int[] nums, int value) {
        HashSet<Integer> reachable = new HashSet<>();
        
        for (int num : nums) {
            int start = num % value;
            if (start < 0) start += value;
            for (int i = 0; i < value; i++) {
                reachable.add(start + i * value);
            }
        }
        
        int mex = 0;
        while (reachable.contains(mex)) {
            mex++;
        }
        
        return mex;
    }
}

File: Medium/2025-10-16-2598-Smallest-Missing-Non-negative-Integer-After-Operations/solution.js
var findSmallestInteger = function(nums, value) {
    const reachable = new Set();
    
    for (const num of nums) {
        const start = num >= 0 ? num % value : (num % value + value) % value;
        for (let i = start; i <= 1e9; i += value) {
            reachable.add(i);
        }
    }
    
    let mex = 0;
    while (reachable.has(mex)) {
        mex++;
    }
    
    return mex;
};